fabric-rendering-v1/src/client/java/net/fabricmc/fabric/impl/client/rendering/hud/HudStatusBarHeightRegistryImpl.java

```diff
@@ -0,0 +1,394 @@
+	 * {@link net.fabricmc.fabric.api.client.rendering.v1.hud.HudStatusBarHeightRegistry#addLeft(Identifier,
+	 * StatusBarHeightProvider)}.
+	 */
+	static final StatusBarHeightProvider ARMOR_BAR = (PlayerEntity player) -> {
+		return player.getArmor() > 0 ? 10 : 0;
+	};
+	/**
+	 * Height provider for the vanilla mount health.
+	 *
+	 * <p>Mods that would otherwise have a mixin for this are encouraged to instead register a replacement provider via
+	 * {@link net.fabricmc.fabric.api.client.rendering.v1.hud.HudStatusBarHeightRegistry#addRight(Identifier,
+	 * StatusBarHeightProvider)}.
+	 */
+	static final StatusBarHeightProvider MOUNT_HEALTH = (PlayerEntity player) -> {
+		InGameHud hud = MinecraftClient.getInstance().inGameHud;
+		LivingEntity livingEntity = ((InGameHudAccessor) hud).callGetRiddenEntity();
+		int vehicleMaxHearts = ((InGameHudAccessor) hud).callGetHeartCount(livingEntity);
+		return ((InGameHudAccessor) hud).callGetHeartRows(vehicleMaxHearts) * 10;
+	};
+	/**
+	 * Height provider for the vanilla food bar.
+	 *
+	 * <p>Mods that would otherwise have a mixin for this are encouraged to instead register a replacement provider via
+	 * {@link net.fabricmc.fabric.api.client.rendering.v1.hud.HudStatusBarHeightRegistry#addRight(Identifier,
+	 * StatusBarHeightProvider)}.
+	 */
+	static final StatusBarHeightProvider FOOD_BAR = (PlayerEntity player) -> {
+		InGameHud hud = MinecraftClient.getInstance().inGameHud;
+		LivingEntity livingEntity = ((InGameHudAccessor) hud).callGetRiddenEntity();
+		return ((InGameHudAccessor) hud).callGetHeartCount(livingEntity) == 0 ? 10 : 0;
+	};
+	/**
+	 * Height provider for the vanilla air bar.
+	 *
+	 * <p>Mods that would otherwise have a mixin for this are encouraged to instead register a replacement provider via
+	 * {@link net.fabricmc.fabric.api.client.rendering.v1.hud.HudStatusBarHeightRegistry#addRight(Identifier,
+	 * StatusBarHeightProvider)}.
+	 */
+	static final StatusBarHeightProvider AIR_BAR = (PlayerEntity player) -> {
+		int maxAirSupply = player.getMaxAir();
+		int airSupply = Math.clamp(player.getAir(), 0, maxAirSupply);
+		boolean isInWater = player.isSubmergedIn(FluidTags.WATER);
+		return isInWater || airSupply < maxAirSupply ? 10 : 0;
+	};
+	/**
+	 * This serves two purposes: it provides a fixed order for some vanilla status bars; and it provides reduced vanilla
+	 * height providers, to compare with the actual height providers during rendering for potential translations for
+	 * vanilla status bars. Translations are achieved via matrix stack transformations; alternatively can also be
+	 * implemented via mixins.
+	 *
+	 * <p>Do not use {@link Map}, it does not preserve insertion order.
+	 */
+	static final Map<Identifier, ResolvedHeightProvider> VANILLA_HEIGHT_PROVIDERS = ImmutableMap.of(VanillaHudElements.HEALTH_BAR,
+			ResolvedHeightProvider.ZERO,
+			VanillaHudElements.ARMOR_BAR,
+			HEALTH_BAR::applyAsInt,
```

Comment:
```suggestion
			HEALTH_BAR,
```
It's already a `ToIntFunction` as far as I can tell

Reply:
This is necessary. The map stores `ResolvedHeightProvider`, `HEALTH_BAR` is a `StatusBarHeightProvider`.